dvc/repo/experiments/run.py

```diff
@@ -42,17 +45,38 @@ def run(
         # Force `_update_params` even if `--set-param` was not used
         path_overrides[hydra_output_file] = []
 
-    if queue:
-        if not kwargs.get("checkpoint_resume", None):
-            kwargs["reset"] = True
-        queue_entry = repo.experiments.queue_one(
-            repo.experiments.celery_queue,
-            targets=targets,
-            params=path_overrides,
-            **kwargs,
+    hydra_sweep = any(
+        x.is_sweep_override()
+        for param_file in path_overrides
+        for x in to_hydra_overrides(path_overrides[param_file])
+    )
+
+    if hydra_sweep and not queue:
+        raise InvalidArgumentError(
+            "Sweep overrides can't be used without `--queue`"
         )
-        name = queue_entry.name or queue_entry.stash_rev[:7]
-        ui.write(f"Queued experiment '{name}' for future execution.")
+
+    if queue:
+        if hydra_sweep:
+            from dvc.utils.hydra import get_hydra_sweeps
+
+            sweeps = get_hydra_sweeps(path_overrides)
+        else:
+            sweeps = [path_overrides]
+
+        for sweep_overrides in sweeps:
+            if not kwargs.get("checkpoint_resume", None):
+                kwargs["reset"] = True
+            queue_entry = repo.experiments.queue_one(
+                repo.experiments.celery_queue,
+                targets=targets,
+                params=sweep_overrides,
+                **kwargs,
+            )
+            if sweep_overrides:
+                ui.write(f"Queueing with overrides '{sweep_overrides}'.")
+            name = queue_entry.name or queue_entry.stash_rev[:7]
+            ui.write(f"Queued experiment '{name}' for future execution.")
         return {}
 
     return repo.experiments.reproduce_one(
```

Comment:
How about an early exit? Something like:

```suggestion

    if not queue:
        return repo.experiments.reproduce_one(
            targets=targets,
            params=path_overrides,
            tmp_dir=tmp_dir,
            **kwargs
        )

    if hydra_sweep:
        from dvc.utils.hydra import get_hydra_sweeps

        sweeps = get_hydra_sweeps(path_overrides)
    else:
        sweeps = [path_overrides]

    for sweep_overrides in sweeps:
        if not kwargs.get("checkpoint_resume", None):
            kwargs["reset"] = True
        queue_entry = repo.experiments.queue_one(
            repo.experiments.celery_queue,
            targets=targets,
            params=sweep_overrides,
            **kwargs,
        )
        if sweep_overrides:
            ui.write(f"Queueing with overrides '{sweep_overrides}'.")
        name = queue_entry.name or queue_entry.stash_rev[:7]
        ui.write(f"Queued experiment '{name}' for future execution.")
    return {}
```